boards/shared/Src/Io/Io_SharedSpi.c

```diff
@@ -41,92 +41,90 @@ void Io_SharedSpi_SetNssHigh(const struct SharedSpi *const spi_interface)
         spi_interface->nss_port, spi_interface->nss_pin, GPIO_PIN_SET);
 }
 
-HAL_StatusTypeDef Io_SharedSpi_TransmitAndReceive(
+bool Io_SharedSpi_TransmitAndReceive(
     const struct SharedSpi *const spi_interface,
     uint8_t *                     tx_buffer,
     uint16_t                      tx_buffer_size,
     uint8_t *                     rx_buffer,
     uint16_t                      rx_buffer_size)
 
 {
+    bool status = false;
+
     Io_SharedSpi_SetNssLow(spi_interface);
-    HAL_StatusTypeDef status = HAL_SPI_Transmit(
-        spi_interface->spi_handle, tx_buffer, tx_buffer_size,
-        spi_interface->timeout_ms);
-    if (status != HAL_OK)
-    {
-        Io_SharedSpi_SetNssHigh(spi_interface);
-        return status;
-    }
 
-    status = HAL_SPI_Receive(
-        spi_interface->spi_handle, rx_buffer, rx_buffer_size,
-        spi_interface->timeout_ms);
-    if (status != HAL_OK)
+    if (HAL_SPI_Transmit(
+            spi_interface->spi_handle, tx_buffer, tx_buffer_size,
+            spi_interface->timeout_ms) == HAL_OK)
     {
-        Io_SharedSpi_SetNssHigh(spi_interface);
-        return (status);
+        status = HAL_SPI_Receive(
+                     spi_interface->spi_handle, rx_buffer, rx_buffer_size,
+                     spi_interface->timeout_ms) == HAL_OK;
     }
+
     Io_SharedSpi_SetNssHigh(spi_interface);
 
-    return HAL_OK;
+    return status;
 }
 
-HAL_StatusTypeDef Io_SharedSpi_Transmit(
+bool Io_SharedSpi_Transmit(
     const struct SharedSpi *const spi_interface,
     uint8_t *                     tx_buffer,
     uint16_t                      tx_buffer_size)
 {
     Io_SharedSpi_SetNssLow(spi_interface);
-    HAL_StatusTypeDef status = HAL_SPI_Transmit(
-        spi_interface->spi_handle, tx_buffer, tx_buffer_size,
-        spi_interface->timeout_ms);
+    const bool status =
+        HAL_SPI_Transmit(
+            spi_interface->spi_handle, tx_buffer, tx_buffer_size,
+            spi_interface->timeout_ms) == HAL_OK;
     Io_SharedSpi_SetNssHigh(spi_interface);
 
     return status;
 }
 
-HAL_StatusTypeDef Io_SharedSpi_Receive(
+bool Io_SharedSpi_Receive(
     const struct SharedSpi *const spi_interface,
     uint8_t *                     rx_buffer,
     uint16_t                      rx_buffer_size)
 {
     Io_SharedSpi_SetNssLow(spi_interface);
-    HAL_StatusTypeDef status = HAL_SPI_Receive(
-        spi_interface->spi_handle, rx_buffer, rx_buffer_size,
-        spi_interface->timeout_ms);
+    const bool status =
+        HAL_SPI_Receive(
+            spi_interface->spi_handle, rx_buffer, rx_buffer_size,
+            spi_interface->timeout_ms) == HAL_OK;
     Io_SharedSpi_SetNssHigh(spi_interface);
 
     return status;
 }
 
-HAL_StatusTypeDef Io_SharedSpi_MultipleTransmitWithoutNssToggle(
+bool Io_SharedSpi_MultipleTransmitWithoutNssToggle(
```

Comment:
I believe this is used once in `Io_LTC6813_ConfigureRegisterA`. Since its only used once i might remove it from `SharedSpi`. I can do this in my PR to refactor LTC6813 related stuff

Reply:
shore